packages/headless/src/controllers/commerce/core/facets/generator/headless-commerce-facet-generator.ts

```diff
@@ -51,13 +52,19 @@ type CommerceFacetBuilder<Facet extends CoreCommerceFacet<any, any>> = (
   options: CommerceFacetOptions
 ) => Facet;
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CommerceSearchableFacetBuilder<Facet extends CoreCommerceFacet<any, any>> =
+  (
+    engine: CommerceEngine,
+    options: CommerceFacetOptions & CommerceSearchableFacetOptions
+  ) => Facet;
 /**
  * @internal
  *
  * The `CommerceFacetGenerator` options used internally.
  */
 export interface CommerceFacetGeneratorOptions {
-  buildRegularFacet: CommerceFacetBuilder<CommerceRegularFacet>;
+  buildRegularFacet: CommerceSearchableFacetBuilder<CommerceRegularFacet>;
   buildNumericFacet: CommerceFacetBuilder<CommerceNumericFacet>;
   buildDateFacet: CommerceFacetBuilder<CommerceDateFacet>;
   // TODO: buildCategoryFacet: CommerceFacetBuilder<CommerceCategoryFacet>;
```

Comment:
Category facets will also be searchable; I'll update the other PR accordingly once this one is merged.